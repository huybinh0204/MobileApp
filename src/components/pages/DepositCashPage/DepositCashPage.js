import React, { useRef } from 'react';
import ENV from 'react-native-config';
import BottomSheet from 'react-native-raw-bottom-sheet';
import { Link, MainButton } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ICONS } from '_constants';
import strings from '_localization/index';
import { openExternalLink } from '_utilities/ExternalLinks';
import { normalize } from '_utilities/screen';
import {
  ContentContainer,
  Description,
  FooterContainer,
  FooterNote,
  FooterNoteLink,
  Illustration,
  SheetContainer,
  SheetContent,
  SheetTitle,
  SheetTnC,
  Title,
} from './DepositCashPage.styles';

const DepositCashPage = () => {
  const infoSheet = useRef(null);

  const openInfoSheet = () => {
    infoSheet.current?.open();
  };

  const closeInfoSheet = () => {
    infoSheet.current?.close();
  };

  const openDepositLocaltions = async () => {
    await openExternalLink(ENV.FIND_PLACE_TO_DEPOSIT_CASH_URI);
  };

  return (
    <SecondaryScreenLayout>
      <ContentContainer>
        <Title>{strings.depositCash.contentTitle}</Title>
        <Illustration
          icon={ICONS.getPaidEarlyIllustration}
          width={normalize(300)}
          height={normalize(300)}
        />
        <Description>{strings.depositCash.contentDescription}</Description>
        <FooterContainer>
          <FooterNote>
            {strings.depositCash.footerNote}
            <FooterNoteLink onPress={openInfoSheet}>
              {strings.depositCash.footerNoteLink}
            </FooterNoteLink>
          </FooterNote>
          <MainButton onPress={openDepositLocaltions}>{strings.depositCash.button}</MainButton>
        </FooterContainer>
      </ContentContainer>
      <BottomSheet height={normalize(420)} ref={infoSheet}>
        <SheetContainer>
          <SheetTitle>{strings.depositCash.infoSheetTitle}</SheetTitle>
          <SheetContent>{strings.depositCash.infoSheetContent}</SheetContent>
          <Link href={ENV.DEPOSIT_CASH_AGREEMENT_URL} wrapperComponent={SheetTnC}>
            {strings.depositCash.infoSheetTermsAndConditions}
          </Link>
          <MainButton onPress={closeInfoSheet}>
            {strings.depositCash.infoSheetButtonLabel}
          </MainButton>
        </SheetContainer>
      </BottomSheet>
    </SecondaryScreenLayout>
  );
};

export default DepositCashPage;
