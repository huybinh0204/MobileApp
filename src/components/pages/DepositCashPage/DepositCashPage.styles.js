import styled from '@emotion/native';
import { BaseText, IconSvg } from '_components/atoms';

export const ContentContainer = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const Title = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading4,
  color: theme.colors.beta900,
  textAlign: 'center',
}));

export const Illustration = styled(IconSvg)(({ theme }) => ({
  alignSelf: 'center',
  marginVertical: theme.spacing.s,
}));

export const Description = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  textAlign: 'center',
}));

export const FooterContainer = styled.View({
  marginTop: 'auto',
});

export const FooterNote = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const FooterNoteLink = styled(BaseText)(({ theme }) => ({
  color: theme.colors.alpha500,
}));

export const SheetContainer = styled.View(({ theme }) => ({
  padding: theme.spacing.m,
}));

export const SheetTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading6,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const SheetContent = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.m,
}));

export const SheetTnC = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.alpha500,
  marginBottom: theme.spacing.m,
}));
