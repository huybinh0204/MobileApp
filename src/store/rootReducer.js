import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
