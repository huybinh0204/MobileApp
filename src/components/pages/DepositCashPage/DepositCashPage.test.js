import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { Linking } from 'react-native';
import ENV from 'react-native-config';
import { createStore } from 'redux';
import strings from '_localization/index';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import DepositCashPage from './DepositCashPage';

describe('DepositCashPage', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = createStore(rootReducer, {});
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', async () => {
    const { toJSON } = render(
      <ComponentWithProviders store={mockStore} component={DepositCashPage} />
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should open the deposit cash locations in external browser', async () => {
    jest.spyOn(Linking, 'openURL');
    const { getByText } = render(
      <ComponentWithProviders store={mockStore} component={DepositCashPage} />
    );

    const findDepositLocationButton = getByText(strings.depositCash.button);
    fireEvent.press(findDepositLocationButton);
    await waitFor(() => {
      expect(Linking.openURL).toHaveBeenCalledWith(ENV.FIND_PLACE_TO_DEPOSIT_CASH_URI);
    });
  });
});
