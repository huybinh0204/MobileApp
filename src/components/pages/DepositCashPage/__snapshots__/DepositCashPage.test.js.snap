// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositCashPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SecondaryScreenLayout"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 43.5,
                                    "lineHeight": 60,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Deposit cash
                            </Text>
                            <SvgMock
                              height={543.5}
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "marginVertical": 29,
                                  },
                                  undefined,
                                ]
                              }
                              width={543.5}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Deposit cash into your account when you visit thousands of Green Dot Network® locations nationwide.*
                            </Text>
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginTop": "auto",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 20,
                                      "lineHeight": 28,
                                      "marginBottom": 29,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                *When you deposit cash, it is transferred by a third party into your account. Additional fees may apply. 
                                <Text
                                  allowFontScaling={false}
                                  onPress={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#3D3397",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Learn more.
                                </Text>
                              </Text>
                              <View
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "opacity": 1,
                                    "paddingHorizontal": 45.5,
                                    "paddingVertical": 31,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  disabled={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Find a location near you
                                </Text>
                              </View>
                            </View>
                          </View>
                          <Modal
                            animationType="none"
                            hardwareAccelerated={false}
                            onRequestClose={[Function]}
                            supportedOrientations={
                              Array [
                                "portrait",
                                "portrait-upside-down",
                                "landscape",
                                "landscape-left",
                                "landscape-right",
                              ]
                            }
                            transparent={true}
                            visible={false}
                          />
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
